### PR TITLE
`azurerm_automation_job_schedule`: update jobSchedule resource id format

### PR DESCRIPTION
**Pull request by wuxu92**
> **Draft**: Wait https://github.com/hashicorp/go-azure-helpers/pull/208 to be released and update the verion in azurerm.

The job schdule is a link between a runbook and a schdule. there should be only one such link between the runbook and schedule pair. When we update a runbook, request a PUT of runbook resource, the job schedule id linked to this runbook will be automatically updated, and the old job schedule id should be NOT FOUND then.

The current implementation introduced in #7555 will try to delete all job schedules linked to the runbook and then it need a next  `terraform apply` to recreate these job schedules. This is not neccessary at first, and it make the `terraform plan` can not report what `terraform apply` really does, it makes users confused when update a runbook.

This PR is to replace the `azurerm_automation_job_schedule` resource id with a customized one. The new one contains the runbook name and the schedule resource name, so we can query the job schdule id in runtime and made the id responded from API as computed attribute `resource_manager_id`. this PR also introduces a schema migration to make sure not break exsiting resources.

the old resource id format: `terraform import azurerm_automation_job_schedule.example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Automation/automationAccounts/account1/jobSchedules/10000000-1001-1001-1001-000000000001`
the new resourec id format: ` /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Automation/automationAccounts/account1/runBook/book1/schedule/schedule1`

resolves:  #8634

```
--- PASS: TestAccAutomationJobSchedule_basic (136.84s)
--- PASS: TestAccAutomationJobSchedule_complete (198.69s)
--- PASS: TestAccAutomationJobSchedule_update (298.63s)
--- PASS: TestAccAutomationJobSchedule_requiresImport (209.73s)
--- PASS: TestAccAutomationRunbook_PSWorkflow (184.43s)
--- PASS: TestAccAutomationRunbook_WithDraft (184.13s)
--- PASS: TestAccAutomationRunbook_requiresImport (193.49s)
--- PASS: TestAccAutomationRunbook_PSWorkflowWithHash (141.74s)
--- PASS: TestAccAutomationRunbook_PSWithContent (186.88s)
--- PASS: TestAccAutomationRunbook_PSWorkflowWithoutUri (177.20s)
--- PASS: TestAccAutomationRunbook_withJobSchedule (543.13s)
--- PASS: TestAccAutomationRunbook_withJobScheduleResourceAndUpdate (211.70s)
```

Test Pass in TC

image

### CODE REVIEW

File: internal/services/automation/automation_runbook_resource_test.go [hunk 1]
```diff
@@ -144,6 +144,31 @@ func TestAccAutomationRunbook_withJobSchedule(t *testing.T) {
 	})
 }
 
+func TestAccAutomationRunbook_withJobScheduleResourceAndUpdate(t *testing.T) {
```
Comment (magodo) [inline internal/services/automation/automation_runbook_resource_test.go:147]: Shall we move this test case to the *automation_job_schedule_resource_test.go* instead, as it is testing that the job schedule won't be impacted by the update of the run book?

File: internal/services/automation/automation_runbook_resource.go [hunk 2]
```diff
@@ -330,31 +341,15 @@ func resourceAutomationRunbookCreateUpdate(d *pluginsdk.ResourceData, meta inter
...
-					return fmt.Errorf("job schedule Id is nil or empty listed by %s Job Schedule List: %+v", id, err)
-				}
-				parsedId := jobschedule.NewJobScheduleID(subscriptionID, id.ResourceGroupName, id.AutomationAccountName, *itemProps.JobScheduleId)
-				if resp, err := jsClient.Delete(ctx, parsedId); err != nil {
-					if !response.WasNotFound(resp.HttpResponse) {
-						return fmt.Errorf("deleting job schedule Id listed by %s Job Schedule List:%v", id, err)
-					}
-				}
-			}
+	// **don't need** to list job schedules and delete all of them. update the runbook will recreate these job schedules automatically,
```
Comment (magodo) [inline internal/services/automation/automation_runbook_resource.go:344]: Comparing to the original implementation, what's the main difference here? From what I see, the list query is different where you have specified an additional filter for only the linked `runbook` name.

File: internal/services/automation/automation_runbook_resource.go [hunk 3]
```diff
@@ -330,31 +341,15 @@ func resourceAutomationRunbookCreateUpdate(d *pluginsdk.ResourceData, meta inter
...
+	// but with a different job schedule id
+	// Fix issue: https://github.com/hashicorp/terraform-provider-azurerm/issues/8634
+	if jsValue, ok := d.GetOk("job_schedule"); (ok && d.IsNewResource()) || d.HasChange("job_schedule") {
```
Comment (magodo) [inline internal/services/automation/automation_runbook_resource.go:347]: Can we just change this to be:

```suggestion
	if d.HasChange("job_schedule") {
```

File: internal/services/automation/automation_runbook_resource.go [hunk 4]
```diff
@@ -299,8 +300,18 @@ func resourceAutomationRunbookCreateUpdate(d *pluginsdk.ResourceData, meta inter
...
+	aaid := jobschedule.NewAutomationAccountID(subscriptionID, id.ResourceGroupName, id.AutomationAccountName)
+	jsIterator, err := jsClient.ListByAutomationAccountComplete(ctx, aaid, jobschedule.ListByAutomationAccountOperationOptions{})
+	if err != nil {
+		return fmt.Errorf("list %v", err)
+	}
+	var item jobschedule.JobSchedule
+	if len(jsIterator.Items) > 0 {
+		item = jsIterator.Items[0]
+		_ = item
+	}
```
Comment (magodo) [inline internal/services/automation/automation_runbook_resource.go:312]: What is the purpose of this part?

File: internal/services/automation/automation_runbook_resource.go [hunk 5]
```diff
@@ -330,31 +341,15 @@ func resourceAutomationRunbookCreateUpdate(d *pluginsdk.ResourceData, meta inter
...
+	// **don't need** to list job schedules and delete all of them. update the runbook will recreate these job schedules automatically,
+	// but with a different job schedule id
+	// Fix issue: https://github.com/hashicorp/terraform-provider-azurerm/issues/8634
+	if jsValue, ok := d.GetOk("job_schedule"); (ok && d.IsNewResource()) || d.HasChange("job_schedule") {
+		if err = deleteLinkedJobSchedules(ctx, jsClient, &id); err != nil {
+			return fmt.Errorf("clean existing job schedule links: %v", err)
 		}
-	}
 
-	if v, ok := d.GetOk("job_schedule"); ok {
```
Comment (magodo) [inline internal/services/automation/automation_runbook_resource.go:356]: The `job_schedule` is not documented in the `azurerm_automation_job_schedule` (apprently was missing in #7555). This is a good chance to add it up. Be note that we shall mention that users should either use the `job_schedule` property in this resource, or use the `azurerm_automation_job_schedule`, but not both. See [this](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/subnet#:~:text=NOTE%20on%20Virtual%20Networks%20and%20Subnet%27s%3A) as a reference.

File: internal/services/automation/automation_runbook_resource_test.go [hunk 6]
```diff
@@ -144,6 +144,31 @@ func TestAccAutomationRunbook_withJobSchedule(t *testing.T) {
...
+			Config: r.withJobScheduleResource(data, "Foo"),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That("azurerm_automation_job_schedule.test").ExistsInAzure(js),
+			),
+		},
+		data.ImportStep("publish_content_link"),
+	})
+}
+
```
Comment (magodo) [inline internal/services/automation/automation_runbook_resource_test.go:171]: Additionally, in the test `TestAccAutomationRunbook_withJobSchedule`, can we test that for a `job_schedule` setting, updating the runbook's other property will works as expected?

File: internal/services/automation/automation_job_schedule_resource_test.go [hunk 7]
```diff
@@ -91,17 +92,17 @@ func TestAccAutomationJobSchedule_requiresImport(t *testing.T) {
 }
 
 func (t AutomationJobScheduleResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
-	id, err := jobschedule.ParseJobScheduleID(state.ID)
+	tfID, err := parse.AutomationJobScheduleID(state.ID)
```
Comment (magodo) [inline internal/services/automation/automation_job_schedule_resource_test.go:95]: Can we rename this variable back to `id` to keep consistency?

File: internal/services/automation/automation_job_schedule_resource.go [hunk 8]
```diff
@@ -246,3 +258,23 @@ func resourceAutomationJobScheduleDelete(d *pluginsdk.ResourceData, meta interfa
 
 	return nil
 }
+
+func GetJobScheduleFromTFID(ctx context.Context, client *jobschedule.JobScheduleClient, tfID *parse.AutomationJobScheduleId) (jsID *jobschedule.JobSchedule, err error) {
```
Comment (magodo) [inline internal/services/automation/automation_job_schedule_resource.go:262]:
```suggestion
func getJobScheduleFromTFID(ctx context.Context, client *jobschedule.JobScheduleClient, id *parse.AutomationJobScheduleId) (js *jobschedule.JobSchedule, err error) {
```

File: internal/services/automation/automation_job_schedule_resource.go [hunk 9]
```diff
@@ -246,3 +258,23 @@ func resourceAutomationJobScheduleDelete(d *pluginsdk.ResourceData, meta interfa
...
+		return nil, fmt.Errorf("loading Automation Account %q Job Schedule List: %+v", tfID.AutomationAccountName, err)
+	}
+
+	var js *jobschedule.JobSchedule
+	for _, item := range jsList.Items {
+		if item.Properties != nil && item.Properties.Runbook != nil &&
+			item.Properties.Runbook.Name != nil && *item.Properties.Runbook.Name == tfID.RunBookName {
+			js = &item
+			break
+		}
```
Comment (magodo) [inline internal/services/automation/automation_job_schedule_resource.go:276]: Probably this check can be codified to the filter with `and`?

File: internal/services/automation/automation_job_schedule_resource.go [hunk 10]
```diff
@@ -185,41 +176,53 @@ func resourceAutomationJobScheduleRead(d *pluginsdk.ResourceData, meta interface
...
+	d.Set("resource_manager_id", js.Id)
 	d.Set("job_schedule_id", id.JobScheduleId)
 	d.Set("resource_group_name", id.ResourceGroupName)
 	d.Set("automation_account_name", id.AutomationAccountName)
 
-	if model := resp.Model; model != nil {
-		if props := model.Properties; props != nil {
-			d.Set("runbook_name", props.Runbook.Name)
-			d.Set("schedule_name", props.Schedule.Name)
+	// list API response lost parameter field, so use Get API to get the JobSchedule
```
Comment (magodo) [inline internal/services/automation/automation_job_schedule_resource.go:204]:
```suggestion
	// The response from the list API has no parameter field, so use Get API to get the JobSchedule
```

File: internal/services/automation/automation_job_schedule_resource.go [hunk 11]
```diff
@@ -232,7 +235,16 @@ func resourceAutomationJobScheduleDelete(d *pluginsdk.ResourceData, meta interfa
 	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := jobschedule.ParseJobScheduleID(d.Id())
+	tfID, err := parse.AutomationJobScheduleID(d.Id())
+	if err != nil {
+		return err
+	}
+	js, err := GetJobScheduleFromTFID(ctx, client, tfID)
+	if err != nil || js == nil {
```
Comment (magodo) [inline internal/services/automation/automation_job_schedule_resource.go:243]: If `js == nil`, shoudl we just return `nil` as it indicates the schedule doesn't exist?

**Review by magodo — CHANGES_REQUESTED** [review]
Thanks for this PR!
I've taken a look through and left some comments inline. The main ask is that this solution is built on an assumption that the schedules linked to a runbook must have different names, would you please help confirm whether this is the case?

**Comment by wuxu92** [comment]
The Schedule name should be unique inside the same automation account, and there MUST be only 1 link between the runbook and the schedule. so the new id schema should work.

File: internal/services/automation/automation_job_schedule_resource.go [hunk 12]
```diff
@@ -185,41 +176,53 @@ func resourceAutomationJobScheduleRead(d *pluginsdk.ResourceData, meta interface
...
+		d.SetId("")
+		return nil
 	}
 
+	id, err := jobschedule.ParseJobScheduleID(pointer.From(js.Id))
+	if err != nil {
+		return fmt.Errorf("parse jobSchedule id %s: %v", pointer.From(js.Id), err)
+	}
+
+	d.Set("resource_manager_id", js.Id)
```
Comment (magodo) [inline internal/services/automation/automation_job_schedule_resource.go:199]: Can we ensure the id is case sensitive by doing:

```suggestion
	d.Set("resource_manager_id", id.ID())
```

File: internal/services/automation/automation_job_schedule_resource.go (hunk 11 above)
Comment (magodo) [reply internal/services/automation/automation_job_schedule_resource.go:243]: I men can we split this to two `if` to make the intent clear (i.e. to avoid being smart)?

```
if err != nil {
  return err
}

if js == nil {
  return nil
}
```

File: internal/services/automation/automation_job_schedule_resource_test.go [hunk 13]
```diff
@@ -91,32 +114,38 @@ func TestAccAutomationJobSchedule_requiresImport(t *testing.T) {
...
 	if err != nil {
 		return nil, fmt.Errorf("retrieving %s: %v", *id, err)
 	}
 
-	return pointer.To(resp.Model != nil), nil
+	return pointer.To(resp != nil), nil
 }
 
-func (AutomationJobScheduleResource) template(data acceptance.TestData) string {
+// add variable parameter runbookDesc, because we need test for update runbook
```
Comment (magodo) [inline internal/services/automation/automation_job_schedule_resource_test.go:130]: Can we remove this comment as the itent is obvious? This kind of comment is more like telling the reviewer the intent of *this* change, while we shall keep the comment to tell stuff to the code viewer at any case.

File: internal/services/automation/automation_runbook_resource.go [hunk 14]
```diff
@@ -549,3 +535,31 @@ func expandDraft(inputs []interface{}) *runbook.RunbookDraft {
...
+	automationAccountId := jobschedule.NewAutomationAccountID(id.SubscriptionId, id.ResourceGroupName, id.AutomationAccountName)
+	filter := fmt.Sprintf("properties/runbook/name eq '%s'", id.RunbookName)
+	jsIterator, err := client.ListByAutomationAccountComplete(ctx, automationAccountId, jobschedule.ListByAutomationAccountOperationOptions{Filter: &filter})
+	if err != nil {
+		return fmt.Errorf("loading Automation Account %q Job Schedule List: %+v", id.AutomationAccountName, err)
+	}
+
+	for _, item := range jsIterator.Items {
+		if itemProps := item.Properties; itemProps != nil {
+			if itemProps.Runbook != nil && itemProps.Runbook.Name != nil && *itemProps.Runbook.Name == id.RunbookName {
```
Comment (magodo) [inline internal/services/automation/automation_runbook_resource.go:549]: Is this already guaranteed by the filter?

File: internal/services/automation/automation_runbook_resource_test.go [hunk 15]
```diff
@@ -124,20 +124,23 @@ func TestAccAutomationRunbook_withJobSchedule(t *testing.T) {
 			Config: r.withJobSchedule(data),
 			Check: acceptance.ComposeTestCheckFunc(
 				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("job_schedule.#").HasValue("1"),
```
Comment (magodo) [inline internal/services/automation/automation_runbook_resource_test.go:127]: This is already covered in the follow-up `ImportStep`, so can be removed

File: internal/services/automation/automation_runbook_resource_test.go [hunk 16]
```diff
@@ -124,20 +124,23 @@ func TestAccAutomationRunbook_withJobSchedule(t *testing.T) {
...
 			),
 		},
 		data.ImportStep("publish_content_link"),
 		{
 			Config: r.withJobScheduleUpdated(data),
 			Check: acceptance.ComposeTestCheckFunc(
 				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("job_schedule.#").HasValue("1"),
```
Comment (magodo) [inline internal/services/automation/automation_runbook_resource_test.go:135]: This is already covered in the follow-up `ImportStep`, so can be removed

File: internal/services/automation/automation_runbook_resource_test.go [hunk 17]
```diff
@@ -124,20 +124,23 @@ func TestAccAutomationRunbook_withJobSchedule(t *testing.T) {
...
 			),
 		},
 		data.ImportStep("publish_content_link"),
 		{
 			Config: r.withoutJobSchedule(data),
 			Check: acceptance.ComposeTestCheckFunc(
 				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("job_schedule.#").HasValue("0"),
```
Comment (magodo) [inline internal/services/automation/automation_runbook_resource_test.go:143]: This is already covered in the follow-up `ImportStep`, so can be removed

File: internal/services/automation/automation_runbook_resource.go (hunk 2 above)
Comment (magodo) [reply internal/services/automation/automation_runbook_resource.go:334]: Can we modify the comment here a bit to state that once the `job_schedule` is changed, recreate them?

File: website/docs/r/automation_runbook.html.markdown [hunk 18]
```diff
@@ -106,7 +106,11 @@ The following arguments are supported:
 
 * `log_activity_trace_level` - (Optional) Specifies the activity-level tracing options of the runbook, available only for Graphical runbooks. Possible values are `0` for None, `9` for Basic, and `15` for Detailed. Must turn on Verbose logging in order to see the tracing.
 
-* `draft` - (Optional) A `draft` block as defined below .
+* `draft` - (Optional) A `draft` block as defined below.
+
+* `job_schedule` - (Optional) Onr or more `job_schedule` block as defined below.
+
+~> **NOTE** AzureRM provides a stand-alone [azurerm_automation_job_schedule](automation_job_schedule.html.markdown) and this inlined `job_schdule` property to manage the job schedules. At this time you should choose one of them to manage the job schedule resources.
```
Comment (magodo) [inline website/docs/r/automation_runbook.html.markdown:113]: Can we also put this notice at the begining of the `automation_job_schedule` document?

File: internal/services/automation/automation_runbook_resource.go [hunk 19]
```diff
@@ -330,31 +331,16 @@ func resourceAutomationRunbookCreateUpdate(d *pluginsdk.ResourceData, meta inter
...
-					if !response.WasNotFound(resp.HttpResponse) {
-						return fmt.Errorf("deleting job schedule Id listed by %s Job Schedule List:%v", id, err)
-					}
-				}
-			}
+	// **don't need** to list job schedules and delete all of them. update the runbook will recreate these job schedules automatically,
+	// but with a different job schedule id
+	// Fix issue: https://github.com/hashicorp/terraform-provider-azurerm/issues/8634
+	if d.HasChange("job_schedule") {
+		jsValue, _ := d.GetOk("job_schedule")
```
Comment (magodo) [inline internal/services/automation/automation_runbook_resource.go:338]:
```suggestion
		jsValue := d.Get("job_schedule")
```

**Review by magodo — CHANGES_REQUESTED** [review]
Thanks for this PR!
I have left some additional mostly minor comments that once addressed this should be good to merge 👍

File: internal/services/automation/resourceids.go [hunk 20]
```diff
@@ -0,0 +1,3 @@
+package automation
+
+//go:generate go run ../../tools/generator-resource-id/main.go -path=./ -name=AutomationJobSchedule -id=/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/group1/providers/Microsoft.Automation/automationAccounts/account1/runBook/book1/schedule/schedule1
```
Comment (tombuildsstuff) [inline internal/services/automation/resourceids.go:3]: The functionality provided by `resourceids.go` shouldn't be used for new resources and will be removed once we've finished migrating off them - instead, since this is already using `hashicorp/go-azure-sdk` - the relevant Resource ID from `hashicorp/go-azure-sdk` should be used instead

Comment (wuxu92) [reply internal/services/automation/resourceids.go:3]: the Resource ID in `go-azure-sdk` cannot work as expect in the issue of this PR is tring to fix. because the job schedule id will be updated automatically when the runbook resource updated. so I create this new job schedule ID Format to represent the link between the runbook and schedule.

Comment (wuxu92) [reply internal/services/automation/resourceids.go:3]: I updated this PR to fix some logic and tested it locally. Although the ID format of the resource has changed, I verified that this update should not introduce any breaking changes.

File: website/docs/r/automation_job_schedule.html.markdown [hunk 21]
```diff
@@ -67,5 +71,5 @@ The `timeouts` block allows you to specify [timeouts](https://www.terraform.io/l
 Automation Job Schedules can be imported using the `resource id`, e.g.
 
 ```shell
-terraform import azurerm_automation_job_schedule.example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Automation/automationAccounts/account1/jobSchedules/10000000-1001-1001-1001-000000000001
+terraform import azurerm_automation_job_schedule.example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Automation/automationAccounts/account1/runBook/book1/schedule/schedule1
```
Comment (katbyte) [inline website/docs/r/automation_job_schedule.html.markdown:74]: Is there a reason we are creating a new "fake" resource id instead of creating a composite resource id? ie
`/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Automation/automationAccounts/account1| /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Automation/automationAccounts/account1/runbooks/Get-AzureVMTutorial`

this is something we are looking to implement first class support for soon and will work with the up coming casing fixes - see https://github.com/hashicorp/go-azure-helpers/pull/208/files

**Review by katbyte — CHANGES_REQUESTED** [review]

File: website/docs/r/automation_job_schedule.html.markdown (hunk 21 above)
Comment (katbyte) [reply website/docs/r/automation_job_schedule.html.markdown:74]: Yes, hopefully it'll be merged in the next couple weeks

Comment (katbyte) [reply website/docs/r/automation_job_schedule.html.markdown:74]: @wuxu92 - the above PR has been merged

Comment (tombuildsstuff) [reply website/docs/r/automation_job_schedule.html.markdown:74]: https://github.com/hashicorp/terraform-provider-azurerm/pull/25446

File: internal/services/automation/automation_job_schedule_resource.go [hunk 22]
```diff
@@ -248,3 +274,18 @@ func resourceAutomationJobScheduleDelete(d *pluginsdk.ResourceData, meta interfa
...
 	return nil
 }
+
+func GetJobScheduleFromTFID(ctx context.Context, client *jobschedule.JobScheduleClient, id *commonids.CompositeResourceID[*schedule.ScheduleId, *runbook.RunbookId]) (js *jobschedule.JobSchedule, err error) {
+	accountID := jobschedule.NewAutomationAccountID(id.First.SubscriptionId, id.First.ResourceGroupName, id.First.AutomationAccountName)
+	filter := fmt.Sprintf("properties/schedule/name eq '%s' and properties/runbook/name eq '%s'", id.First.ScheduleName, id.Second.RunbookName)
+	jsList, err := client.ListByAutomationAccountComplete(ctx, accountID, jobschedule.ListByAutomationAccountOperationOptions{Filter: &filter})
+	if err != nil {
+		return nil, fmt.Errorf("loading Automation Account %q Job Schedule List: %+v", accountID.AutomationAccountName, err)
+	}
```
Comment (tombuildsstuff) [inline internal/services/automation/automation_job_schedule_resource.go:284]: why is this needed? The Go SDK has a `Get` method?

```go
func (c RunbookClient) Get(ctx context.Context, id RunbookId) (result GetOperationResponse, err error) {
```

File: internal/services/automation/resourceids.go [hunk 23]
```diff
@@ -0,0 +1,3 @@
+package automation
```
Comment (tombuildsstuff) [inline internal/services/automation/resourceids.go:1]: can we remove this file?

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/automation/automation_job_schedule_resource.go [hunk 24]
```diff
@@ -107,43 +125,24 @@ func resourceAutomationJobScheduleCreate(d *pluginsdk.ResourceData, meta interfa
 		jobScheduleUUID = uuid.FromStringOrNil(jobScheduleID.(string))
 	}
 
-	id := jobschedule.NewJobScheduleID(subscriptionId, d.Get("resource_group_name").(string), d.Get("automation_account_name").(string), jobScheduleUUID.String())
+	// accountID := automationaccount.NewAutomationAccountID(subscriptionId, resourceGroup, accountName)
```
Comment (catriona-m) [inline internal/services/automation/automation_job_schedule_resource.go:128]: should this be removed?

File: internal/services/automation/automation_job_schedule_resource.go [hunk 25]
```diff
@@ -187,41 +187,53 @@ func resourceAutomationJobScheduleRead(d *pluginsdk.ResourceData, meta interface
...
-			d.SetId("")
-			return nil
-		}
-		return fmt.Errorf("making Read request on %s: %+v", *id, err)
+		return err
+	}
+	if js == nil {
+		d.SetId("")
+		return nil
 	}
```
Comment (catriona-m) [inline internal/services/automation/automation_job_schedule_resource.go:203]: In the case where the resource group has been deleted outside of terraform, this will continually give us an error when trying to list the job schedules and it is never marked as gone. We should update this to cover this case.

File: website/docs/r/automation_runbook.html.markdown [hunk 26]
```diff
@@ -106,7 +106,11 @@ The following arguments are supported:
 
 * `log_activity_trace_level` - (Optional) Specifies the activity-level tracing options of the runbook, available only for Graphical runbooks. Possible values are `0` for None, `9` for Basic, and `15` for Detailed. Must turn on Verbose logging in order to see the tracing.
 
-* `draft` - (Optional) A `draft` block as defined below .
+* `draft` - (Optional) A `draft` block as defined below.
+
+* `job_schedule` - (Optional) Onr or more `job_schedule` block as defined below.
```
Comment (catriona-m) [inline website/docs/r/automation_runbook.html.markdown:111]:
```suggestion
* `job_schedule` - (Optional) One or more `job_schedule` block as defined below.
```

File: internal/services/automation/automation_job_schedule_resource.go [hunk 27]
```diff
@@ -279,6 +278,9 @@ func GetJobScheduleFromTFID(ctx context.Context, client *jobschedule.JobSchedule
 	filter := fmt.Sprintf("properties/schedule/name eq '%s' and properties/runbook/name eq '%s'", id.First.ScheduleName, id.Second.RunbookName)
 	jsList, err := client.ListByAutomationAccountComplete(ctx, accountID, jobschedule.ListByAutomationAccountOperationOptions{Filter: &filter})
 	if err != nil {
+		if response.WasNotFound(jsList.LatestHttpResponse) {
+			return nil, nil
+		}
```
Comment (catriona-m) [inline internal/services/automation/automation_job_schedule_resource.go:283]: I tested this and it seems that the response can sometimes be nil causing this to fail

**Review by catriona-m — CHANGES_REQUESTED** [review]
Thanks for making the changes @wuxu92 - I tested this and still got an error in some cases, could you take another look? Thanks!

File: internal/services/automation/automation_job_schedule_resource.go (hunk 27 above)
Comment (catriona-m) [reply internal/services/automation/automation_job_schedule_resource.go:283]: > I apologize for not fully testing the app. I believe this issue is caused by an SDK problem and should be resolved in Pandora. I have created a PR to fix it at [hashicorp/pandora#4129](https://github.com/hashicorp/pandora/pull/4129). If possible, could you please review it?

Thanks @wuxu92 - this is now merged 👍

File: website/docs/r/automation_job_schedule.html.markdown [hunk 28]
```diff
@@ -67,5 +71,5 @@ The `timeouts` block allows you to specify [timeouts](https://www.terraform.io/l
 Automation Job Schedules can be imported using the `resource id`, e.g.
 
 ```shell
-terraform import azurerm_automation_job_schedule.example /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Automation/automationAccounts/account1/jobSchedules/10000000-1001-1001-1001-000000000001
+terraform import azurerm_automation_job_schedule.example "/subscriptions/85b3dbca-5974-4067-9669-67a141095a76/resourceGroups/acctestRG-auto-240326135902419761/providers/Microsoft.Automation/automationAccounts/acctestAA-240328091346495956/schedules/acctestAS-240328091346495956|/subscriptions/85b3dbca-5974-4067-9669-67a141095a76/resourceGroups/acctestRG-auto-240326135902419761/providers/Microsoft.Automation/automationAccounts/acctestAA-240328091346495956/runbooks/Output-HelloWorld"
```
Comment (catriona-m) [inline website/docs/r/automation_job_schedule.html.markdown:74]:
```suggestion
terraform import azurerm_automation_job_schedule.example "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Automation/automationAccounts/account1/schedules/schedule1|/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.Automation/automationAccounts/account1/runbooks/runbook1"
```

File: internal/services/automation/automation_job_schedule_resource.go [hunk 29]
```diff
@@ -187,41 +186,53 @@ func resourceAutomationJobScheduleRead(d *pluginsdk.ResourceData, meta interface
...
+		return err
+	}
+	if js == nil {
+		d.SetId("")
+		return nil
+	}
+
+	id, err := jobschedule.ParseJobScheduleID(pointer.From(js.Id))
+	if err != nil {
+		return fmt.Errorf("parse jobSchedule id %s: %v", pointer.From(js.Id), err)
```
Comment (catriona-m) [inline internal/services/automation/automation_job_schedule_resource.go:206]:
```suggestion
		return err
```

**Review by catriona-m — CHANGES_REQUESTED** [review]
Thanks for fixing this up @wuxu92 - I spotted a couple of minor things on looking through this again but otherwise this is looking good. Thanks!

**Review by catriona-m — APPROVED** [review]
Thanks for making the additional changes @wuxu92 - LGTM!
